### PR TITLE
OSD-7210: Move AccountClaim Validation into the AccountClaim controller

### PR DESCRIPTION
**Pull request by mrWinston**
This PR makes sure, that the accountclaim validation happens before any account is even created by moving the validation code to `pkg/controller/accountclaim/accountclaim_controller.go`.

Additional unit tests are also added to cover the basics of creating a byoc accountclaim.

### CODE REVIEW

File: hack/templates/aws.managed.openshift.io_v1alpha1_ccs_accountclaim_cr_invalid.tmpl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+apiVersion: v1
```
Comment (dkeohane) [inline hack/templates/aws.managed.openshift.io_v1alpha1_ccs_accountclaim_cr_invalid.tmpl:1]: Care to leave a comment at the top of this file indicating how it's an invalid accountclaim?

Comment (mrWinston) [reply hack/templates/aws.managed.openshift.io_v1alpha1_ccs_accountclaim_cr_invalid.tmpl:1]: I would just go ahead and remove this file and associated make targets again, as i don't really see the need for this kind of integration test now that we have the unit test properly set up.

Comment (dkeohane) [reply hack/templates/aws.managed.openshift.io_v1alpha1_ccs_accountclaim_cr_invalid.tmpl:1]: Sounds good :+1:

File: pkg/controller/accountclaim/accountclaim_controller_test.go [hunk 2]
```diff
@@ -142,6 +136,57 @@ var _ = Describe("AccountClaim", func() {
...
+		When("Accountclaim is BYOC", func() {
+
+			BeforeEach(func() {
+				accountClaim.SetFinalizers(append(accountClaim.GetFinalizers(), accountClaimFinalizer))
+				accountClaim.Spec.BYOC = true
+				accountClaim.Spec.AccountLink = ""
+			})
+
+			It("should fail validation", func() {
+				// this should cause Validate() to fail
```
Comment (drpaneas) [inline pkg/controller/accountclaim/accountclaim_controller_test.go:149]: Did I get that right?

```suggestion
				// fail validation if BYOC is not associated with an account
```

**Review by drpaneas — APPROVED** [review]
/lgtm it seems legit change, relocating this codeblock to the appropriate place.

File: pkg/controller/accountclaim/accountclaim_controller_test.go (hunk 2 above)
Comment (mrWinston) [reply pkg/controller/accountclaim/accountclaim_controller_test.go:149]: Yep, that's better!

**Review by NautiluX — COMMENTED** [review]
/lgtm

**Pull request merged**